tests/sentry/relay/config/test_metric_extraction.py

```diff
@@ -1227,3 +1585,48 @@ def test_get_metric_extraction_config_with_no_spec(default_project: Project) ->
         assert config
         assert len(config["metrics"]) == 1
         assert config["metrics"][0].get("condition") is None
+
+
+def _metric_spec(tags: list[dict[str, str]]) -> dict[str, Any]:
+    return {
+        "category": "transaction",
+        "condition": {"name": "event.duration", "op": "gte", "value": 10.0},
+        "field": None,
+        "mri": "c:transactions/on_demand@none",
+        "tags": tags,
+    }
+
+
+@django_db_all
+def test_include_environment_for_widgets(default_project: Project) -> None:
+    aggr = "count()"
+    env_tag = {"field": "event.environment", "key": "environment"}
+    query = "transaction.duration:>=10"
+
+    with Feature([ON_DEMAND_METRICS, ON_DEMAND_METRICS_WIDGETS]):
+        create_widget([aggr], query, default_project)
+        config = get_metric_extraction_config(default_project)
+        # Because we have two specs we will have two metrics.
+        # The second spec includes the environment tag as part of the query hash.
+        # XXX: Verify that the query hash is as expected
+        assert config and config["metrics"] == [
+            _metric_spec([{"key": "query_hash", "value": "f1353b0f"}, env_tag]),
+            _metric_spec([{"key": "query_hash", "value": "4fb5a472"}, env_tag]),
+        ]
+
+        # Using environment in the columns will turn into a groupbys
+        create_widget([aggr], query, default_project, columns=["environment"], title="foo")
+        config = get_metric_extraction_config(default_project)
+        assert config and config["metrics"] == [
+            _metric_spec([{"key": "query_hash", "value": "f1353b0f"}, env_tag]),
+            _metric_spec([{"key": "query_hash", "value": "4fb5a472"}, env_tag]),
+        ]
+
+        create_alert(aggr, query, default_project)
+        config = get_metric_extraction_config(default_project)
+        # XXX: Ask Kev about this
```

Comment:
@k-fish I think we should always prefer the widget over the alert. What do you think?

Reply:
hmm, a `widget` and `alert` with the new spec version should have different query hashes so they should no longer collide.

Can we remove the `columns=["environment"]` from the widget and test this for spec-version(1)? 

The result when testing an identical alert and widget (without an environment column in either) should be two separate query hashes, since the specs should be different. 

You'll have identical specs _except_ the widget spec will have an `environment` tag which is automatically added because of `DYNAMIC_QUERY`, and with spec version 1 that tag being there should also cause a different hash.